src/Workspaces/Core/Portable/LanguageServices/DeclaredSymbolFactoryService/AbstractDeclaredSymbolInfoFactoryService.cs

```diff
@@ -229,10 +239,11 @@ private void AddDeclaredSymbolInfos(
                 cancellationToken);
 
             // Then recurse into the children and add those.
+            var receiverTypeName = this.GetExtensionReceiverTypeName(typeDeclaration);
```

Comment:
Should we check for `null` before going into `foreach` loop below?

Reply:
def not.  this is getting the receiver name *if* we are hitting an extension block.  but we always want to create the child infos regardless of what thsi returns.  it's just that if this is an extension, we get teh receiver type, and pass that in so that when we create the infos for the members we know what extension type to associate them with.